diff --git a/scripts/setlocalversion b/scripts/setlocalversion
index e80fd64535d1..fec3f163f05c 100755
--- a/scripts/setlocalversion
+++ b/scripts/setlocalversion
@@ -107,12 +107,6 @@ scm_version()
 		# it fails. Note that git-diff-index does not refresh the
 		# index, so it may give misleading results. See
 		# git-update-index(1), git-diff-index(1), and git-status(1).
-		if {
-			git --no-optional-locks status -uno --porcelain 2>/dev/null ||
-			git diff-index --name-only HEAD
-		} | read dummy; then
-			printf '%s' -dirty
-		fi
 	fi
 }
 
